RecoLocalCalo/HGCalRecProducers/plugins/HGCalCLUEAlgo.cc

```diff
@@ -216,89 +231,253 @@ math::XYZPoint HGCalCLUEAlgo::calculatePosition(const std::vector<int>& v, const
+          int binId = lt.getGlobalBinByBin(xBin, yBin);
+          size_t binSize = lt[binId].size();
+
+          for (unsigned int j = 0; j < binSize; j++) {
+            unsigned int otherId = lt[binId][j];
+            bool otherSi;
+            if (isOnlySi) {
+              otherSi = true;
+            } else if (cellsOnLayer.isSi[otherId]) {
+              otherSi = true;
+            } else {
+              otherSi = false;
+            }
+            if (otherSi) {  //silicon cells cannot talk to scintillator cells
+              if (distance(i, otherId, layerId, false) < delta) {
+                cellsOnLayer.rho[i] += (i == otherId ? 1.f : 0.5f) * cellsOnLayer.weight[otherId];
+              }
+            }
           }
         }
       }
+    } else {
+      delta = delta_r;
+      std::array<int, 4> search_box = lt.searchBoxEtaPhi(cellsOnLayer.eta[i] - delta,
+                                                         cellsOnLayer.eta[i] + delta,
+                                                         cellsOnLayer.phi[i] - delta,
+                                                         cellsOnLayer.phi[i] + delta);
+      cellsOnLayer.rho[i] += cellsOnLayer.weight[i];
+      float northeast(0), northwest(0), southeast(0), southwest(0), all(0);
+
+      for (int etaBin = search_box[0]; etaBin < search_box[1] + 1; ++etaBin) {
+        for (int phiBin = search_box[2]; phiBin < search_box[3] + 1; ++phiBin) {
+          int binId = lt.getGlobalBinByBinEtaPhi(etaBin, phiBin);
+          size_t binSize = lt[binId].size();
+
+          for (unsigned int j = 0; j < binSize; j++) {
+            unsigned int otherId = lt[binId][j];
+            if (!cellsOnLayer.isSi[otherId]) {  //scintillator cells cannot talk to silicon cells
+              if (distance(i, otherId, layerId, true) < delta) {
+                // need to round these numbers to 5 decimals to be able to compare floats
+                constexpr float roundDec = 100000.;
+                float iPhi = round(cellsOnLayer.phi[i] * roundDec) / roundDec;
+                float otherphi = cellsOnLayer.phi[otherId];
+                otherphi += (otherphi * iPhi >= 0. || abs(iPhi) < 1.) ? 0. : iPhi > 0. ? 2 * M_PI : -2 * M_PI;
+                float otherPhi = round(otherphi * roundDec) / roundDec;
+                float otherEta = round(cellsOnLayer.eta[otherId] * roundDec) / roundDec;
+                float iEta = round(cellsOnLayer.eta[i] * roundDec) / roundDec;
+                LogDebug("HGCalCLUEAlgo") << "  Debugging calculateLocalDensity for Scintillator: \n"
+                                          << "    cell: " << otherId << " energy: " << cellsOnLayer.weight[otherId]
+                                          << " otherPhi: " << otherPhi << " iPhi: " << iPhi << " otherEta: " << otherEta
+                                          << " iEta: " << iEta << "\n";
+
+                if (otherId != i) {
+                  all += 0.5f * cellsOnLayer.weight[otherId];
```

Comment:
please define a temp ```auto offset = 0.5f * cellsOnLayer.weight[otherId];```
the compiler likely did it already, but it should be easier to read

Reply:
Done.